it/repositories/OnlineTestRepositorySpec.scala

```diff
@@ -198,7 +197,77 @@ class OnlineTestRepositorySpec extends MongoRepositorySpec {
         onlineTestRepo.nextExpiringApplication.futureValue must be(None)
       }
     }
+  }
+
+  "nextTestForReminder" should {
+    "return one result" when {
+      "there is an application in PHASE1_TESTS and is about to expiry in the next 72 hours" in {
+        val date = DateTime.now().plusHours(70)
+        val testProfile = Phase1TestProfile(expirationDate = date, tests = List(phase1Test))
+        createApplicationWithAllFields(UserId, AppId, "frameworkId", "SUBMITTED")
+        onlineTestRepo.insertOrUpdatePhase1TestGroup(AppId, testProfile).futureValue
+        onlineTestRepo.nextTestForReminder(FirstReminder).futureValue must be (Some(NotificationExpiringOnlineTest(AppId,UserId,"Georgy", date)))
+        onlineTestRepo.nextTestForReminder(SecondReminder).futureValue must be(None)
+      }
+
+      "there is an application in PHASE1_TESTS and is about to expiry in the next 24 hours" in {
+        val date = DateTime.now().plusHours(22)
+        val testProfile = Phase1TestProfile(expirationDate = date, tests = List(phase1Test))
+        createApplicationWithAllFields(UserId, AppId, "frameworkId", "SUBMITTED")
+        onlineTestRepo.insertOrUpdatePhase1TestGroup(AppId, testProfile).futureValue
+        onlineTestRepo.nextTestForReminder(SecondReminder).futureValue must be (Some(NotificationExpiringOnlineTest(AppId,UserId,"Georgy",date)))
+      }
+    }
+
+    "return no results" when {
+      val date = DateTime.now().plusHours(22)
+      val testProfile = Phase1TestProfile(expirationDate = date, tests = List(phase1Test))
+      "there are no application in PHASE1_TESTS" in {
+        createApplicationWithAllFields(UserId, AppId, "frameworkId", "SUBMITTED")
+        onlineTestRepo.insertOrUpdatePhase1TestGroup(AppId, testProfile).futureValue
+        updateApplication(BSONDocument("applicationStatus" -> ApplicationStatus.IN_PROGRESS))
```

Comment:
I had a look at it and it shouldn't but I'll double check before pushing the next patch ;-)
Perhaps I can make it more explicit as from what i can remember it was waiting  inside the createApplicationWithAllFields for the future to complete which is not probably really clear from the method's signature.